app/src/main/java/com/example/contactlistapp/MainActivity.kt

```diff
@@ -1,101 +1,120 @@
+        // Set up the search functionality using the SearchView
+        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                return false
+            }
 
+            override fun onQueryTextChange(newText: String?): Boolean {
+                filter(newText ?: "")
+                return true
             }
-        }.setNegativeButton("Cancel") { _, _ ->
-            Toast.makeText(this, "Cancel", Toast.LENGTH_LONG).show()
-        }
-        .create().show()
+        })
+
+
+        // Set up the "Add New Contact" button click listene
+        binding.btAddNewContact.setOnClickListener {
+            // Inflate the dialog layout using data binding
+            val inflater = LayoutInflater.from(this)
+
+
+            val dialogBinding: AddnewcontactBinding = DataBindingUtil.inflate(
+                inflater,
+                R.layout.addnewcontact,
+                null,
+                false
+            )
+            dialogBinding.vm = viewModel
+            dialogBinding.lifecycleOwner = this
+
+            val nameBB = dialogBinding.etName
+            val emailBB = dialogBinding.etEmail
+            val numberBB = dialogBinding.etNumber
+
+            // Show an AlertDialog for adding new contact
+            AlertDialog.Builder(this)
+                .setView(dialogBinding.root)
+                .setPositiveButton("OK") { dialog, _ ->
+                    val name = nameBB.text.toString()
+                    val number = numberBB.text.toString()
+                    val email = emailBB.text.toString()
+
+                    //var namevm = ContactProfileData(name,number, email)
+                    if (name.isNotEmpty() && number.isNotEmpty() && email.isNotEmpty()) {
+                        userList.add(
+                            ContactProfileData(
+                                "Name: $name",
+                                "Mobile No. : $number",
+                                email
+                            )
+                        )
+                        adapter.notifyDataSetChanged()
+                        Toast.makeText(this, "Adding User Information Success", Toast.LENGTH_SHORT)
+                            .show()
+                        dialog.dismiss()
+                        //viewModel.saveData()
+                        Toast.makeText(this, "Adding contact", Toast.LENGTH_LONG).show()
+                    } else {
+                        Toast.makeText(this, "Please fill in all fields", Toast.LENGTH_SHORT).show()
+
+                    }
+                }.setNegativeButton("Cancel") { _, _ ->
+                    Toast.makeText(this, "Cancel", Toast.LENGTH_LONG).show()
+                }
+                .create().show()
 
-}
+        }
         // Initialize the ViewModel using ViewModelProvider
         viewModel = ViewModelProvider(this)[ContactProfileViewModel::class.java]
```

Comment:
You have already initialized the viewModel in the onCreate method.
You can remove this initialization as it's redundant.